src/query/storage/index.go

```diff
@@ -220,7 +225,15 @@ func matcherToQuery(matcher models.Matcher) (idx.Query, error) {
 		negate = true
 		fallthrough
 	case models.MatchRegexp:
-		query, err := idx.NewRegexpQuery(matcher.Name, matcher.Value)
+		var (
+			query idx.Query
+			err   error
+		)
+		if bytes.Equal(dotStar, matcher.Value) {
+			query = idx.NewFieldQuery(matcher.Name)
```

Comment:
Do we also support negation just by placing this code here?

Reply:
we should get it for free based on line 240 
```golang
		if negate {
			query = idx.NewNegationQuery(query)
		}
```

Reply:
Great, looks good then.